plugins/source/src/main/kotlin/app/aaps/plugins/source/PathedSyaiTagAppPlugin.kt

```diff
@@ -0,0 +1,95 @@
+
+@Singleton
+class PathedSyaiTagAppPlugin @Inject constructor(injector: HasAndroidInjector, rh: ResourceHelper, aapsLogger: AAPSLogger, )
+    : PluginBase(PluginDescription()
+        .mainType(PluginType.BGSOURCE)
+        .fragmentClass(BGSourceFragment::class.java.name)
+        .preferencesId(R.xml.pref_bgsource)
+        .pluginIcon(app.aaps.core.ui.R.mipmap.ic_syai_tag)
+        .pluginName(R.string.patched_syai_tag_app)
+        .description(R.string.description_source_patched_syai_tag_app),
+    aapsLogger, rh, injector), BgSource {
+
+    // cannot be inner class because of needed injection
+    class PathedSyaiTagAppWorker(context: Context, params: WorkerParameters) : LoggingWorker(context, params, Dispatchers.IO) {
+
+        @Inject lateinit var mSyaiTagAppPlugin: PathedSyaiTagAppPlugin
+        @Inject lateinit var injector: HasAndroidInjector
+        @Inject lateinit var dateUtil: DateUtil
+        @Inject lateinit var repository: AppRepository
+
+        init {
+            (context.applicationContext as HasAndroidInjector).androidInjector().inject(this)
+        }
+
+        override suspend fun doWorkAndLog(): Result {
+            var ret = Result.success()
+            if (!mSyaiTagAppPlugin.isEnabled()) return Result.success()
+            val collection = inputData.getString("collection") ?: return Result.failure(workDataOf("Error" to "missing collection"))
+            if (collection == "entries") {
+                val data = inputData.getString("data")
+                aapsLogger.debug(LTag.BGSOURCE, "Received SyaiTag App Data: $data")
+                if (!data.isNullOrEmpty()) {
+                    try {
+                        val glucoseValues = mutableListOf<TransactionGlucoseValue>()
+                        val jsonArray = JSONArray(data)
+                        for (i in 0 until jsonArray.length()) {
+                            val jsonObject = jsonArray.getJSONObject(i)
+                            when (val type = jsonObject.getString("type")) {
+                                "sgv" ->{
+                                    glucoseValues += TransactionGlucoseValue(
+                                        timestamp = jsonObject.getLong("date"),
+                                        value = jsonObject.getDouble("sgv"),
+                                        raw = jsonObject.getDouble("sgv"),
+                                        noise = null,
+                                        trendArrow = GlucoseValue.TrendArrow.fromString(jsonObject.getString("direction")),
+                                        sourceSensor = GlucoseValue.SourceSensor.SyaiTagApp)
+                                    }
+                                    else  -> aapsLogger.debug(LTag.BGSOURCE, "Unknown entries type: $type")
+                                }
+                            }
+                            repository.runTransactionForResult(CgmSourceTransaction(glucoseValues, emptyList(), null))
```

Comment:
Change the way of DB insertion like it's in dev now. Ie. https://github.com/nightscout/AndroidAPS/blob/dev/plugins/source/src/main/kotlin/app/aaps/plugins/source/PoctechPlugin.kt#L77